forgerock-openbanking-uk-aspsp-common/src/test/java/com/forgerock/openbanking/common/services/frequency/FrequencyServiceTest.java

```diff
@@ -0,0 +1,210 @@
+        // Given
+        DateTime firstDateTime = DateTime.now();
+        String frequency = FrequencyType.EVERYDAY.getFrequencyStr() + "x";
+
+        // When
+        frequencyService.getNextDateTime(firstDateTime, frequency);
+    }
+
+    // EVERYWORKINGDAY("EvryWorkgDay"),
+    @Test
+    public void shouldMatchPattern_everyWorkingDay() {
+        // Given
+        DateTime firstDateTime = DateTime.now();
+        String frequency = FrequencyType.EVERYWORKINGDAY.getFrequencyStr();
+
+        // When
+        DateTime dateTime = frequencyService.getNextDateTime(firstDateTime, frequency);
+
+        // Then
+        assertThat(dateTime).isNotNull();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldRaiseException_everyWorkingDay() {
+        // Given
+        DateTime firstDateTime = DateTime.now();
+        String frequency = FrequencyType.EVERYWORKINGDAY.getFrequencyStr() + "x";
+
+        // When
+        frequencyService.getNextDateTime(firstDateTime, frequency);
+    }
+
+    // INTERVALWEEKDAY("IntrvlWkDay", "0?([1-9]):0?([1-7])$")
+    @Test
+    public void shouldMatchPattern_weekday() {
+        // Given
+        DateTime firstDateTime = DateTime.now();
+        String frequency = FrequencyType.INTERVALWEEKDAY.getFrequencyStr() + ":01:07";
+
+        // When
+        DateTime dateTime = frequencyService.getNextDateTime(firstDateTime, frequency);
+
+        // Then
+        assertThat(dateTime).isNotNull();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldRaiseException_weekday() {
+        // Given
+        DateTime firstDateTime = DateTime.now();
+        String frequency = FrequencyType.INTERVALWEEKDAY.getFrequencyStr() + ":00:07";
+
+        // When
+        frequencyService.getNextDateTime(firstDateTime, frequency);
+    }
+
+    // WEEKINMONTHDAY("WkInMnthDay", "0?([1-5]):0?([1-7])$")
+    @Test
+    public void shouldMatchPattern_weekInMonthDay() {
+        // Given
+        DateTime firstDateTime = DateTime.now();
+        String frequency = FrequencyType.WEEKINMONTHDAY.getFrequencyStr() + ":02:03";
+
+        // When
+        DateTime dateTime = frequencyService.getNextDateTime(firstDateTime, frequency);
+
+        // Then
+        assertThat(dateTime).isNotNull();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldRaiseException_weekInMonthDay() {
+        // Given
+        DateTime firstDateTime = DateTime.now();
+        String frequency = FrequencyType.WEEKINMONTHDAY.getFrequencyStr() + ":00:03";
+
+        // When
+        frequencyService.getNextDateTime(firstDateTime, frequency);
+    }
+    // INTERVALMONTHDAY("IntrvlMnthDay", "(0?[1-6]|12|24):(-0?[1-5]|0?[1-9]|[12][0-9]|3[01])$")
+    @Test
+    public void shouldMatchPattern_intervalMonthDay() {
+        // Given
+        DateTime firstDateTime = DateTime.now();
+        String frequency = FrequencyType.INTERVALMONTHDAY.getFrequencyStr() + ":01:30";
```

Comment:
I'm not familiar with this format - I guess `30` means the 30th of each month, but what does `01` mean? 

I see it comes back as: `2021-03-30T15:47:59.989+01:00` (GMT+1 due to your location).

Does the 01 mean it should start in one month's time?! 😕

Reply:
https://openbanking.atlassian.net/wiki/spaces/DZ/pages/937623689/Domestic+Standing+Orders+v3.1